src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -509,7 +509,7 @@ class TypeApplications(val self: Type) extends AnyVal {
    *               { type $hkArg$0 = T1; ...; type $hkArg$n = Tn }
    *
    *  satisfies predicate `p`. Try base types in the order of their occurrence in `baseClasses`.
```

Comment:
The documentation still seems incomplete, we don't always check that there is a base type where the type parameters match `tparams`, and there's a special case when `tparams.isEmpty`.